src/lib.rs

```diff
@@ -264,32 +264,51 @@ impl<'text> InitialInfoExt<'text> {
         text: &'a str,
         default_para_level: Option<Level>,
     ) -> InitialInfoExt<'a> {
-        let (original_classes, paragraphs, pure_ltr) =
-            compute_initial_info(data_source, text, default_para_level);
+        let (original_classes, paragraphs, pure_ltr, _, _) =
+            compute_initial_info(data_source, text, default_para_level, true);
 
         InitialInfoExt {
             base: InitialInfo {
                 text,
                 original_classes,
-                paragraphs,
+                paragraphs: paragraphs.unwrap(),
             },
-            pure_ltr,
+            pure_ltr: pure_ltr.unwrap(),
         }
     }
 }
 
 /// Implementation of initial-info computation.
+/// This is used for both BidiInfo and ParagraphBidiInfo, with split_paragraphs=true
+/// indicating that it should split paragraphs and return the two Option<> results
+/// (for BidiInfo), and split_paragraphs=false indicating it should not split paras,
+/// and returns the single paragraph_level and is_pure_ltr values instead.
```

Comment:
issue: I don't want the callers to have to unwrap (we want to avoid unwraps as much as possible in unicode_bidi)

can split_paragraphs be implemented as an `Option<&mut Vec>` outparam instead?

(might need to use an option of a tuple to support `pure_ltr` as well)

Reply:
I'll have a look and see what I can come up with...